File: 7-reverse-integer/7-reverse-integer.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        temp = []
        if x < 0:
            sign = -1
        else:
            sign = 1
            
        x = abs(x)
        while x:
            temp.append(x%10)
            x = x//10
            
        temp = temp[::-1]
        res = 0
        for i in range(len(temp)):
            res+=temp[i]*pow(10,i)
        
        res = res*sign
        
        if (res > (pow(2,31)-1)) or (res < pow(-2,31)):
            return 0
        else:
            return res

File: 7-reverse-integer/7-reverse-integer.java
import java.util.*;
class Solution {
    public int reverse(int x) {
        String s = Integer.toString(x);
        ArrayList<Integer> temp = new ArrayList<Integer>(s.length());
        int sign  = 0;
        if (x<0)
            sign = -1;
        else
            sign = 1;
            
        x = Math.abs(x);
        
        while(x>=1){
            temp.add(x%10);
            x = x/10;
        }
        
        Collections.reverse(temp);
        
        int result = 0;
        
        for (int i = 0; i<temp.size(); i++){
            result += temp.get(i)*Math.pow(10,i);
        }
        
        result = result * sign;
        
        if ((result >= (Math.pow(2,31)-1)) || (result <= (Math.pow(-2,31)+1))){
            return 0;
        }
        else{
            return result;
        }
    }
}